sympy/solvers/solvers.py

```diff
@@ -375,6 +375,73 @@ def checksol(f, symbol, sol=None, **flags):
     # returns None if it can't conclude
     # TODO: improve solution testing
 
+def _is_eq_bool(eq):
+    """Returns true if Eq or Ne has True or False as an argument"""
+    if isinstance(eq, (Equality, Unequality)):
+        return any(side in (S.false, S.true) for side in eq.args)
+
+def _invert_eq_bool(fi):
+    """Returns the expression when one argument is a bool and the other an expression in an Eq or Ne"""
+    if isinstance(fi, (Equality, Unequality)):
+        lhs, rhs = fi.args
+        unequal = S.true if isinstance(fi, Unequality) else S.false
+        if lhs in (S.false, S.true) and rhs.is_Relational:
+            return ~rhs if lhs is unequal else rhs
+        if rhs in (S.false, S.true) and lhs.is_Relational:
+            return ~lhs if rhs is unequal else lhs
+    return fi
+
+def _handle_relational(f_relational, symbols, flags):
+    if flags.get('dict', False):
+        solution = reduce_inequalities(f_relational, symbols=symbols)
+        if isinstance(solution, Equality):
+            return [{solution.lhs: solution.rhs}]
+        solution = list(solution.args)
+        for sol in solution:
+            # Behavior for types like StrictLessThan is not defined
+            if not isinstance(sol, (Or, Equality)):
+                warnings.warn(filldedent('''
+                    Warning: Ignoring dict=True due to
+                    incompatible type of %s.''' % sol))
+                return reduce_inequalities(f_relational, symbols=symbols)
+        intermediate = [sol for sol in solution
+                        if isinstance(sol, Equality)]
+        solution[:] = [sol.args for sol in solution
+                       if sol not in intermediate]
+        solutions = [intermediate + list(sol)
+                     for sol in itertools.product(*solution)]
+        for i in range(len(solutions)):
+            ele = {}
+            for sol in solutions[i]:
+                ele[sol.lhs] = sol.rhs
+            solutions[i] = ele
+        return [] if solutions == [{}] else solutions
+    return reduce_inequalities(f_relational, symbols=symbols)
+
+def _collapse_eq_to_expr(fi):
+    if not isinstance(fi, Equality):
+        return fi
+    elif 'ImmutableDenseMatrix' in [type(a).__name__ for a in fi.args]:
+        return fi.lhs - fi.rhs
+    else:
+        return fi.rewrite(Add, evaluate=False)
+
+def _swap_dict(solution, swap_sym):
+    """Helper function to handle pure dict-like solutions like {y(t): 2, x(t): 1/3}"""
+    solution = {swap_sym.get(k, k): v.subs(swap_sym)
+                     for k, v in solution.items()}
+    return solution
```

Comment:
I don't think that factoring out one line functions such as this always helps to clarify the code. It can do if the code is confusing and the function name makes the effect clearer than the code itself but `_swap_dict` (and the docstring here) are not any more informative than the code itself.